src/FormControl/test/FormControlSpec.tsx

```diff
@@ -550,4 +550,35 @@ describe('FormControl', () => {
       });
     });
   });
+
+  describe('ruleDependencies', () => {
+    it('Should validate the field when the dependency changes', () => {
```

Comment:
Form provides a way to cross-check dependencies between fields.
https://github.com/rsuite/schema-typed?tab=readme-ov-file#multi-field-cross-validation

If you want to add validation rules through `rule`, you can use `Schema.Types.StringType().when(...)`

https://github.com/rsuite/schema-typed?tab=readme-ov-file#whencondition-schemaspec-schemadeclarationdatatype-errormsgtype--type